### PR TITLE
Fixes int64, and uint8 test cases for CUDA 

### PR DESCRIPTION
**Pull request by hmccarty**
New to the code base, but this fixed int64 and uint8 test cases for me using CUDA

Relevant test cases added here: #866

### CODE REVIEW

File: tinygrad/runtime/ops_cuda.py [hunk 1]
```diff
@@ -46,8 +46,10 @@ def __call__(self, global_size, local_size, *args, wait=False):
 
 class CUDACodegen(CStyleCodegen):
   lang = CStyleLanguage(
-    kernel_prefix = "__global__", smem_prefix = "__shared__ ", barrier = "__syncthreads();", float4 = "make_float4",
+    kernel_prefix = "using int64 = int64_t; using uchar = unsigned char;\n__global__", smem_prefix = "__shared__ ",
+    barrier = "__syncthreads();", float4 = "make_float4",
     half_prekernel = "#include <cuda_fp16.h>",
+    float_conversion = {x.name: "__float2uint_rn" if dtypes.is_unsigned(x) else "__float2int_rn" for x in [dtypes.int8, dtypes.int32, dtypes.int64, dtypes.uint8]},
```
Comment (dc-dc-dc) [inline tinygrad/runtime/ops_cuda.py:52]: can dtypes.is_int() be used here ?

Comment (hmccarty) [reply tinygrad/runtime/ops_cuda.py:52]: Came up `{x.name: "__float2uint_rn" if dtypes.is_unsigned(x) else "__float2int_rn" for x in map(lambda x: x.default, fields(dtypes)) if dtypes.is_int(x)}`, but requires another line for import

I say we leave it, as similar direct reference is also done in `llvmir.py`. If need arises, maybe an `all_dtypes` function will be added in the future.

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/codegen/cstyle.py                 3               0
tinygrad/runtime/ops_cuda.py               4               2
------------------------------------------------------------
total                                      7               2
------------------------------------------------------------
lines added in the tinygrad folder: 5
```

**Comment by tinyb0t** [comment]
Changes made in `tinygrad/`:
```
------------------------------------------------------------
files                             insertions       deletions
------------------------------------------------------------
tinygrad/codegen/cstyle.py                 3               0
tinygrad/runtime/ops_cuda.py               4               2
------------------------------------------------------------
total                                      7               2
------------------------------------------------------------
lines added in the tinygrad folder: 5
```

**Comment by geohot** [comment]
We need CUDA in CI, otherwise these PRs are really hard to merge. There's a $200 bounty for doing it.

Check out https://github.com/gpuocelot/gpuocelot

**Comment by hmccarty** [comment]
Fixed in https://github.com/geohot/tinygrad/pull/962

**Pull request closed without merging**